TestingHomework-Discounts/Accessors/ProductAdminAccessor.cs

```diff
@@ -0,0 +1,61 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using TestingHomework.Tests.Data;
+using TestingHomework_Discounts;
+
+namespace TestingHomework.Tests.Accessors
+{
+    public interface IProductAdminAccessor
+    {
+        IEnumerable<Product> GetAllProducts();
+
+        Product SaveProduct(Product product);
+
+        Product GetProduct(Guid productId);
+    }
+    public class ProductAdminAccessor : IProductAdminAccessor
+    {
+
+
+        Product_Mapper mapper = new Product_Mapper();
```

Comment:
This class properly uses mapping between Domain Objects and DTOs. Only domain objects are accepted and returned and only DTOs are used with the database.